src/shogun/machine/gp/ExactInferenceMethod.cpp

```diff
@@ -199,7 +199,7 @@ void CExactInferenceMethod::update_mean()
 	m_mu=SGVector<float64_t>(m.vlen);
 	Map<VectorXd> eigen_mu(m_mu.vector, m_mu.vlen);
 
-	eigen_mu=eigen_K*CMath::sq(m_scale)*eigen_alpha;
+	eigen_mu=eigen_K*CMath::exp(m_log_scale*2.0)*eigen_alpha;
```

Comment:
it seems
`CMath::sq(get_scale())` has two function calls while  `exp(m_log_scale*2.0)` has one call.